Yank/yank.py

```diff
@@ -202,17 +266,55 @@ def __setstate__(self, serialization):
         self._topology = mdtraj.Topology.from_dataframe(atoms, bonds)
         self._ligand_atoms = serialization['ligand_atoms']
         self._solvent_atoms = serialization['solvent_atoms']
+        self._regions = serialization['regions']
 
     # -------------------------------------------------------------------------
     # Internal-usage
     # -------------------------------------------------------------------------
 
-    def _resolve_atom_indices(self, atoms_description):
-        if isinstance(atoms_description, str):
+    def _check_existing_regions(self, region_string):
+        """Make sure regions don't overlap"""
+        if region_string in self:
+            raise KeyError("{} is already part of this Topology! "
+                           "Cannot overwrite built-in regions!".format(region_string))
+
+    @property
+    def _combined_regions(self):
+        """
+        Return the combined set of regions
+        This is its own property despite its simplicity since several functions and methods call it
+        """
+        # Iterate over the dict_keys object since its a static view and should be thread-safe
+        return self._built_in_regions + [region for region in self._regions.keys()]
+
+    def __contains__(self, item):
+        """Check the in operator to see if region is in this class"""
+        return item in self._combined_regions
```

Comment:
This is fine since `_combined_regions` should be very small. A cleaner alternative here could be
```python
return item in self._regions or item in self._BUILT_IN_REGIONS
```
This way the search would be based on the hash instead of instantiating and searching through a list.

Reply:
Okay, there was another place I had `_combined_regions` but I suppose I can put them into the same `in` comparison now that I have implementation done.